test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -1322,6 +1322,171 @@ bool VerifyIconPathExists_Unpackaged()
     return true;
 }
 
+bool VerifyRegisterWithNullDisplayNameFail_Unpackaged()
+{
+    // Register is already called in main with an explicit appusermodelId
+    winrt::AppNotificationManager::Default().UnregisterAll();
+    try
+    {
+        winrt::AppNotificationManager::Default().Register(winrt::hstring{}, winrt::Uri{ LR"(C:\Windows\System32\WindowsSecurityIcon.png)" });
+
+
+        winrt::AppNotificationManager::Default().UnregisterAll();
+    }
+    catch (...)
+    {
+        return winrt::to_hresult() == E_INVALIDARG;
+    }
+
+    return false;
+}
+
+bool VerifyRegisterWithNullIconFail_Unpackaged()
+{
+    // Register is already called in main with an explicit appusermodelId
+    winrt::AppNotificationManager::Default().UnregisterAll();
+    try
+    {
+        winrt::AppNotificationManager::Default().Register(L"AppNotificationApp", nullptr);
+
+        winrt::AppNotificationManager::Default().UnregisterAll();
```

Comment:
A better pattern in this situation, would be to call  UnregisterAll() in a scope_exit. It would
1) make it instantly clear that UnregisterAll isn't under test here.
2) though it doesn't apply to this specific case as Unregister can't throw E_INVALIDARG, avoid a situation where the fuction under test does not throw as expected but the cleanup one does. The test would then pass although it should be failing.